powerstore/resource_protection_policy.go

```diff
@@ -269,7 +297,11 @@ func (r resourceProtectionPolicy) Update(ctx context.Context, req tfsdk.UpdateRe
 	log.Printf("Successfully done with Update")
 }
 
-func (r resourceProtectionPolicy) planToProtectionPolicyParam(plan models.ProtectionPolicy) *gopowerstore.ProtectionPolicyCreate {
+func (r resourceProtectionPolicy) planToProtectionPolicyParam(plan models.ProtectionPolicy) (protectionPolicyCreate *gopowerstore.ProtectionPolicyCreate, errmsg string) {
```

Comment:
instead of errmsg string,
best practice will be to return error
and the code, where you are using error as string, u may use err.Error()

Reply:
Made requested changes